platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpoint.java

```diff
@@ -159,73 +155,65 @@ public Response create(MultipartBody multipartBody, @Context UriInfo requestUriI
    * @param attachment the attachment to copy and extract.
    * @return The file of the copied attachment.
    */
+  @Nullable
   private File createFileFromAttachement(Attachment attachment) {
-    InputStream inputStream = null;
-    String filename = null;
-    File newFile = null;
-    if (attachment.getContentDisposition() != null) {
-      filename =
-          attachment
-              .getContentDisposition()
-              .getParameter(FILENAME_CONTENT_DISPOSITION_PARAMETER_NAME);
-    }
-
-    if (StringUtils.isEmpty(filename)) {
-      LOGGER.debug("Filename not found, using default.");
-      filename = DEFAULT_FILE_NAME;
-    } else {
-      filename = FilenameUtils.getName(filename);
-      LOGGER.debug("Filename: {}", filename);
-    }
+    try (InputStream inputStream = attachment.getDataHandler().getInputStream()) {
+      if (inputStream == null) {
```

Comment:
❓ Will this only be null when an Exception is thrown?

According to the java docs on `DataHandler.getInputStream`:
`If [finding a DataContentHandler for the DataHandler] is successful [if it isn't successful it will throw an exception], it creates a pipe and a thread. The thread uses the DataContentHandler's writeTo method to write the stream data into one end of the pipe. The other end of the pipe is returned to the caller. Because a thread is created to copy the data, IOExceptions that may occur during the copy can not be propagated back to the caller. The result is an empty stream.`

Java docs on `DataSource.getInputStream` say:
`This method returns an InputStream representing the data and throws the appropriate exception if it can not do so.`